apps/condo/domains/meter/hooks/useCreateMeterModal.tsx

```diff
@@ -1,93 +1,47 @@
-import { FormattedMessage } from 'react-intl'
-import { IMeterResourceUIState } from '../utils/clientSchema/MeterResource'
-import { Modal } from 'antd'
-import { IMeterFormState } from '../utils/clientSchema/Meter'
+import { BaseMeterModalForm } from '../components/BaseMeterModal/BaseMeterModalForm'
+import { Meter } from '../utils/clientSchema'
+import { Typography } from 'antd'
 
-type MeterInfoModalTitleProps = {
-    resourceId: string
-}
-
-const MeterInfoModalTitle = ({ resourceId }: MeterInfoModalTitleProps) => {
-    const intl = useIntl()
-    const ResourceTitle = intl.formatMessage({ id: resourceIdToCreateMeterTitleIdMap[resourceId] })
-
-    return (
-        <FormattedMessage
-            id='pages.condo.meter.AddMeterModalTitleTemplate'
-            values={{
-                resource: ResourceTitle,
-            }}
-        />
-    )
-}
-
-type CreateMeterModalProps = {
-    addMeterToFormAction: (defaultValue?: StoreValue, insertIndex?: number) => void
-    resources: IMeterResourceUIState[]
-    newMeters: IMeterFormState[]
-}
-
-export const useCreateMeterModal = ()=> {
+export function useCreateMeterModal (organizationId: string, propertyId: string, unitName: string, refetch) {
     const intl = useIntl()
-    const ChooseServiceMessage = intl.formatMessage({ id: 'pages.condo.meter.ChooseService' })
-    const AddMessage = intl.formatMessage({ id: 'Add' })
+    const AddMeterMessage = intl.formatMessage({ id: 'pages.condo.meter.AddMeter' })
 
     const [isCreateMeterModalVisible, setIsCreateMeterModalVisible] = useState<boolean>(false)
-    const [selectedMeterResourceId, setSelectedMeterResourceId] = useState<string | null>(null)
-
-    useEffect(() => {
-        setSelectedMeterResourceId(null)
-    }, [isCreateMeterModalVisible])
-
-    const CreateMeterModal = ({ addMeterToFormAction, resources, newMeters }: CreateMeterModalProps) => {
-        return !selectedMeterResourceId ? (
-            <Modal
-                title={ChooseServiceMessage}
-                visible={isCreateMeterModalVisible}
-                centered={true}
-                footer={[]}
-                onCancel={() => setIsCreateMeterModalVisible(false)}
-            >
-                <ResourcesList
-                    resources={resources}
-                    setSelectedMeterResource={setSelectedMeterResourceId}
-                />
-            </Modal>
-        ) : (
-            <BaseModalForm
+    const createMeterAction = Meter.useCreate(
+        {},
+        () => {
+            refetch()
+        })
+
+    const handleMeterCreate = useCallback(values => {
+        const numberOfTariffs = values.numberOfTariffs ? values.numberOfTariffs : 1
+        createMeterAction({ ...values, numberOfTariffs, organization: organizationId, property: propertyId, unitName })
+        setIsCreateMeterModalVisible(false)
+    },
+    [createMeterAction, organizationId, propertyId, unitName])
```

Comment:
what about  createMeterAction, values, numberOfTariffs, organizationId, propertyId, unitName? setIsCreateMeterModalVisible ? :)

Reply:
createMeterAction, propertyId, unitName there are, values it's a function argument, numberOfTariffs it's a constant inside this function,